view_user_birthdate.py

```diff
@@ -0,0 +1,242 @@
+        username = first_line.group('username')
+    else:
+        username = None
+
+    # Parse nationality.
+    second_line = re.search("nationality:"
+                            "(?P<nationality>[a-zA-Z]+ ?[a-zA-Z]*)", data_chunk[1])
+    if second_line:
+        nationality = second_line.group('nationality')
+    else:
+        nationality = None
+
+    # Parse birthdate.
+    last_line = re.search("birthdate:"
+                          "(?P<birthdate>[0-9]{1,2}/[0-9]{1,2}/[0-9]{4})", data_chunk[2])
+    if last_line:
+        birthdate = last_line.group('birthdate')
+    else:
+        birthdate = None
+
+    return username, nationality, birthdate
+
+def build_bday_book(file_name):
+    """Take in the name of a file. Use the contents of the file to
+    build a dictionary where the keys of the dictionary are the
+    usernames and the values are the corresponding birthdates.
+    """
+    data = read_file(file_name)
+    assert data, "There is no data!"
+    data_chunks = chunk_data(data, 3)
+    for chunk in data_chunks:
+        username, nationality, birthdate = parse(chunk)
+        if username and nationality and birthdate:
+            month, day, year = build_date(birthdate)
+            birthdate = format_date(month, day, year, nationality)
+            # Instantiate a User object thus adding the User to the bday_book.
+            User(username, nationality, birthdate)
+    return User.bday_book
+
+def build_date(string):
+    """Take in a string containing a date in the format mm/dd/yyyy and
+    return three strings representing a valid day, month, and year.
+    """
+    regex_rule = ("(?P<month>[0-9]{1,2})" # Month.
+                  "/(?P<day>[0-9]{1,2})" # Day.
+                  "/(?P<year>[0-9]{4})" # Year.
+                 )
+    date = re.search(regex_rule, string)
+    assert date, "A string in the format mm/dd/yyyy must be provided."
+
+    month = make_int(date.group('month'))
+    day = make_int(date.group('day'))
+    year = make_int(date.group('year'))
+
+    if is_valid_date(month, day, year):
+        return str(month), str(day), str(year)
+
+def format_date(month, day, year, nationality):
+    """Take in four strings 'month', 'day', 'year' and 'nationality'.
+    Use the given nationality to determine what type of Date object to
+    return. Return None if there is not a nationality match.
+    """
+    assert valid_month(make_int(month)), "Invalid month."
+    assert valid_day(make_int(day), make_int(month)), "Invalid day."
+    assert valid_year(make_int(year)), "Invalid year."
+    assert re.search("[a-zA-Z]+ ?[a-zA-Z]*", nationality), "Invalid nationality."
+    if nationality.lower() in AmericanDate.nationalities:
+        return AmericanDate(month, day, year).format()
+    if nationality.lower() in EuropeanDate.nationalities:
+        return EuropeanDate(month, day, year).format()
+    if nationality.lower() in ChineseDate.nationalities:
+        return ChineseDate(month, day, year).format()
```

Comment:
consider holding on to the actual date objects longer, and only format them when you need to. You may need to add other functionality to date later, and that'd require a refactor since right now you're passing around strings everywhere instead of the object they derived from.